subsys/net/ip/net_mgmt.c

```diff
@@ -386,10 +386,17 @@ void net_mgmt_event_init(void)
 	(void)memset(events, 0, CONFIG_NET_MGMT_EVENT_QUEUE_SIZE *
 			sizeof(struct mgmt_event_entry));
 
+#if IS_ENABLED(CONFIG_NET_TC_THREAD_COOPERATIVE)
+/* Lowest priority cooperative thread */
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1)
+#else
+#define THREAD_PRIORITY K_PRIO_PREEMPT(CONFIG_NUM_PREEMPT_PRIORITIES - 1)
+#endif
+
 	k_thread_create(&mgmt_thread_data, mgmt_stack,
 			K_KERNEL_STACK_SIZEOF(mgmt_stack),
 			(k_thread_entry_t)mgmt_thread, NULL, NULL, NULL,
-			CONFIG_NET_MGMT_EVENT_THREAD_PRIO, 0, K_NO_WAIT);
+			THREAD_PRIORITY, 0, K_NO_WAIT);
```

Comment:
There is a predefined `K_LOWEST_APPLICATION_THREAD_PRIO` in `include/kernel.h` which could be used here.

Reply:
I did this change same way as the other networking threads are done so not using this define. The name of the define talks about application and we have no applications here but a thread which is part of a subsystem. But it is certainly possible to use that define if needed, but I would do that change in a separate PR if we really want to do it.

Reply:
From the kernel perspective subsystem is an application. There is no thread priority range that is reserved for subsystems. So the name `K_LOWEST_APPLICATION_THREAD_PRIO` carries in fact the right meaning. Though I agree that seeing such a define in a networking stack might be confusing. From the user perspective subsystem is certainly not an application. Using the define was just a suggestion.